### ISSUE TITLE
Watch mode triggers a new build for each file changed

### ISSUE THREAD

**Issue by slikts** [dotansimha/graphql-code-generator#1827]
`gql-gen --watch` triggers a build per each source file changed, meaning that, for example, checking out a different git branch with N differing sources will also start N build processes.

Reproduction:

```yml
documents: "*.graphql"
generates:
  output.txt:
    plugins:
      - time
```
```sh
gql-gen --watch
touch foo{1,2,3}.graphql
```
The output will show generation running three times in a row.

The simplest fix should be that if a generation is in progress and a file changes, only one additional generation should be scheduled. Ideally it should also be possible to cancel the current generation.

**Comment by slikts** [comment]
On second thought, the simplest fix would be just debouncing the generator function.

**Comment by dotansimha** [comment]
Added debounce in: https://github.com/dotansimha/graphql-code-generator/pull/2359

**Comment by dotansimha** [comment]
Fixed in 1.6.0.
